config/env.go

```diff
@@ -140,3 +141,136 @@ func bindLegacyClientEnv() {
 		}
 	}
 }
+
+// bindClassAdConfig reads configuration from the HTCondor job ClassAd file.
+// It looks for attributes prefixed with "PelicanCfg_" and converts them to Pelican configuration.
+// For example, "PelicanCfg_Client_PreferredCaches" becomes "Client.PreferredCaches".
+// This function should be called from InitClient after InitConfigInternal.
+func bindClassAdConfig() {
+	// Look for the condor job ad file
+	condorJobAd, isPresent := os.LookupEnv("_CONDOR_JOB_AD")
+	if !isPresent {
+		// No job ad file specified, nothing to do
+		return
+	}
+
+	// Open the job ad file
+	file, err := os.Open(condorJobAd)
+	if err != nil {
+		log.Debugf("Unable to open job ad file %s: %v", condorJobAd, err)
+		return
+	}
+	defer file.Close()
+
+	// Parse the ClassAd using old-style reader (newline-delimited format)
+	reader := classad.NewOldReader(file)
+	if !reader.Next() {
+		if err := reader.Err(); err != nil {
+			log.Debugf("Unable to parse job ad file %s: %v", condorJobAd, err)
+		}
+		return
+	}
+
+	ad := reader.ClassAd()
+	if ad == nil {
+		log.Debugf("No ClassAd found in job ad file %s", condorJobAd)
+		return
+	}
+
+	// Iterate through all attributes in the ClassAd looking for PelicanCfg_ prefix
+	for _, attrName := range ad.GetAttributes() {
+		// Skip attributes that don't have the PelicanCfg_ prefix
+		if !strings.HasPrefix(attrName, "PelicanCfg_") {
+			continue
+		}
+
+		// Convert PelicanCfg_Client_PreferredCaches -> Client.PreferredCaches
+		viperKey := strings.Replace(strings.TrimPrefix(attrName, "PelicanCfg_"), "_", ".", -1)
+
+		// Evaluate the attribute to get its value
+		attrValue := ad.EvaluateAttr(attrName)
+		if attrValue.IsUndefined() {
+			log.Debugf("ClassAd attribute %s is undefined, skipping", attrName)
+			continue
+		}
+
+		// Set the value in viper based on the type
+		if attrValue.IsString() {
+			if strVal, err := attrValue.StringValue(); err == nil {
+				viper.Set(viperKey, strVal)
+				log.Debugf("Set %s = %s from job ClassAd attribute %s", viperKey, strVal, attrName)
+			} else {
+				log.Debugf("Failed to get string value for ClassAd attribute %s: %v", attrName, err)
+			}
+		} else if attrValue.IsInteger() {
+			if intVal, err := attrValue.IntValue(); err == nil {
+				viper.Set(viperKey, intVal)
+				log.Debugf("Set %s = %d from job ClassAd attribute %s", viperKey, intVal, attrName)
+			} else {
+				log.Debugf("Failed to get integer value for ClassAd attribute %s: %v", attrName, err)
+			}
+		} else if attrValue.IsBool() {
+			if boolVal, err := attrValue.BoolValue(); err == nil {
+				viper.Set(viperKey, boolVal)
+				log.Debugf("Set %s = %t from job ClassAd attribute %s", viperKey, boolVal, attrName)
+			} else {
+				log.Debugf("Failed to get boolean value for ClassAd attribute %s: %v", attrName, err)
+			}
+		} else if attrValue.IsReal() {
```

Comment:
Add a test covering the case where an integer or real number is given where a bool is expected.  Does it work?